Fixes compilation error.

The header /usr/include/netinet/in.h on Solaris defines macro 's_addr' as
'_S_un._S_addr' to hide implementation details.  The macro then however,
conflicts with a variable of the same name in Samba's source code.

diff -rup samba-4.18.6.orig/lib/tsocket/tsocket_bsd.c samba-4.18.6/lib/tsocket/tsocket_bsd.c
--- samba-4.18.6.orig/lib/tsocket/tsocket_bsd.c	2023-09-12 13:06:47.988389572 +0200
+++ samba-4.18.6/lib/tsocket/tsocket_bsd.c	2023-09-12 13:08:02.522312530 +0200
@@ -539,7 +539,7 @@ int _tsocket_address_inet_from_hostport_
 	char *buf = NULL;
 	uint64_t port = 0;
 	int ret;
-	char *s_addr = NULL;
+	char *_s_addr = NULL;
 	uint16_t s_port = default_port;
 	bool conv_ret;
 	bool is_ipv6_by_squares = false;
@@ -568,7 +568,7 @@ int _tsocket_address_inet_from_hostport_
 		/* IPv6 possibly with port - squares detected */
 		port_sep = pr_sq + 1;
 		if (*port_sep == '\0') {
-			s_addr = pl_sq + 1;
+			_s_addr = pl_sq + 1;
 			*pr_sq = 0;
 			s_port = default_port;
 			goto get_addr;
@@ -590,7 +590,7 @@ int _tsocket_address_inet_from_hostport_
 		s_port = (uint16_t)port;
 		*port_sep = 0;
 		*pr_sq = 0;
-		s_addr = pl_sq + 1;
+		_s_addr = pl_sq + 1;
 		*pl_sq = 0;
 		goto get_addr;
 	} else if (pl_period != NULL && port_sep != NULL) {
@@ -607,18 +607,18 @@ int _tsocket_address_inet_from_hostport_
 		}
 		s_port = (uint16_t)port;
 		*port_sep = 0;
-		s_addr = buf;
+		_s_addr = buf;
 		goto get_addr;
 	} else {
 		/* Everything else, let tsocket_address_inet_from string() */
 		/* find parsing errors */
-		s_addr = buf;
+		_s_addr = buf;
 		s_port = default_port;
 		goto get_addr;
 	}
 get_addr:
 	ret = _tsocket_address_inet_from_strings(
-	    mem_ctx, fam, s_addr, s_port, _addr, location);
+	    mem_ctx, fam, _s_addr, s_port, _addr, location);
 
 	return ret;
 }
